From 83ab5adcde62be3b4a545dd8e11d5b6eae84f490 Mon Sep 17 00:00:00 2001
From: Andrei Cherechesu <andrei.cherechesu@nxp.com>
Date: Thu, 2 Mar 2023 13:38:13 +0200
Subject: [PATCH 43/54] Revert "s32g3: rdb3: Enable SAR_ADC early clock via
 SCMI"

This reverts commit bb119c1df6767cb0779f7eb2863de44b2db7bcb4.

The ADC support in ATF is deleted as it's no longer needed for
identifying the RDB revision to apply a DDR fixup.

Issue: ALB-9722
Upstream-Status: Pending 

Signed-off-by: Andrei Cherechesu <andrei.cherechesu@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 plat/nxp/s32/include/s32_clocks.h                  |  1 -
 plat/nxp/s32/s32_bl2_scp.c                         | 11 ++---------
 plat/nxp/s32/s32g/s32g3/s32g399ardb3/platform.mk   |  5 ++---
 .../s32g399ardb3/s32g399ardb3_scp_early_clocks.c   | 14 --------------
 4 files changed, 4 insertions(+), 27 deletions(-)
 delete mode 100644 plat/nxp/s32/s32g/s32g3/s32g399ardb3/s32g399ardb3_scp_early_clocks.c

diff --git a/plat/nxp/s32/include/s32_clocks.h b/plat/nxp/s32/include/s32_clocks.h
index 069b8891e..5eeac5331 100644
--- a/plat/nxp/s32/include/s32_clocks.h
+++ b/plat/nxp/s32/include/s32_clocks.h
@@ -160,7 +160,6 @@ enum s32g_mc_cgm {
 
 int s32_plat_clock_init(bool skip_ddr_clk);
 int s32_scp_plat_clock_init(bool skip_ddr_clk);
-int scp_scmi_clk_set_config_enable(unsigned int clock_index);
 int s32_enable_ddr_clock(void);
 int s32_enable_a53_clock(void);
 int scp_reset_ddr_periph(void);
diff --git a/plat/nxp/s32/s32_bl2_scp.c b/plat/nxp/s32/s32_bl2_scp.c
index 5b0d191f0..e8b4e732f 100644
--- a/plat/nxp/s32/s32_bl2_scp.c
+++ b/plat/nxp/s32/s32_bl2_scp.c
@@ -21,13 +21,6 @@
 #include <dt-bindings/clock/s32gen1-scmi-clock.h>
 #include <dt-bindings/reset/s32gen1-scmi-reset.h>
 
-#pragma weak scp_enable_board_early_clocks
-
-int scp_enable_board_early_clocks(void)
-{
-	return 0;
-}
-
 static int scp_scmi_reset_set_state(uint32_t domain_id, bool assert)
 {
 	int ret;
@@ -71,7 +64,7 @@ static int scp_scmi_reset_set_state(uint32_t domain_id, bool assert)
 	return 0;
 }
 
-int scp_scmi_clk_set_config_enable(unsigned int clock_index)
+static int scp_scmi_clk_set_config_enable(unsigned int clock_index)
 {
 	int ret;
 	unsigned int token = 0;
@@ -253,7 +246,7 @@ int s32_scp_plat_clock_init(bool skip_ddr_clk)
 	if (!skip_ddr_clk)
 		return scp_enable_ddr_clock();
 
-	return scp_enable_board_early_clocks();
+	return 0;
 }
 
 int scp_reset_ddr_periph(void)
diff --git a/plat/nxp/s32/s32g/s32g3/s32g399ardb3/platform.mk b/plat/nxp/s32/s32g/s32g3/s32g399ardb3/platform.mk
index a1567d23f..63a832828 100644
--- a/plat/nxp/s32/s32g/s32g3/s32g399ardb3/platform.mk
+++ b/plat/nxp/s32/s32g/s32g3/s32g399ardb3/platform.mk
@@ -1,5 +1,5 @@
 #
-# Copyright 2022 NXP
+# Copyright 2022-2023 NXP
 #
 # SPDX-License-Identifier: BSD-3-Clause
 #
@@ -10,7 +10,6 @@ S32_BOARD_PATH		:= ${PLAT_SOC_PATH}/s32g399ardb3
 
 PLAT_INCLUDES		+= -I${S32_BOARD_PATH}/include \
 
-PLAT_BL_COMMON_SOURCES  += ${S32_BOARD_PATH}/s32g399ardb3_ddr_fixup.c \
-		${S32_BOARD_PATH}/s32g399ardb3_scp_early_clocks.c \
+PLAT_BL_COMMON_SOURCES  += ${S32_BOARD_PATH}/s32g399ardb3_ddr_fixup.c
 
 DTB_FILE_NAME		?= s32g399a-rdb3.dtb
diff --git a/plat/nxp/s32/s32g/s32g3/s32g399ardb3/s32g399ardb3_scp_early_clocks.c b/plat/nxp/s32/s32g/s32g3/s32g399ardb3/s32g399ardb3_scp_early_clocks.c
deleted file mode 100644
index ada50f537..000000000
--- a/plat/nxp/s32/s32g/s32g3/s32g399ardb3/s32g399ardb3_scp_early_clocks.c
+++ /dev/null
@@ -1,14 +0,0 @@
-// SPDX-License-Identifier: BSD-3-Clause
-/*
- * Copyright 2022 NXP
- */
-
-#include <s32_bl_common.h>
-#include <s32_clocks.h>
-
-#include <dt-bindings/clock/s32gen1-scmi-clock.h>
-
-int scp_enable_board_early_clocks(void)
-{
-	return scp_scmi_clk_set_config_enable(S32GEN1_SCMI_CLK_SAR_ADC_BUS);
-}
-- 
2.25.1

